tests/pipelines/test_pipelines_fill_mask.py

```diff
@@ -302,7 +302,9 @@ def run_test_targets(self, model, tokenizer):
         )
         target_ids = {vocab[el] for el in targets}
         self.assertEqual({el["token"] for el in outputs}, target_ids)
-        self.assertEqual({el["token_str"] for el in outputs}, set(targets))
+        # For some BEP tokenizers, `</w>` is removed during decoding, so `token_str` won't be the same as in `targets`.
+        _targets = [x if not x.endswith("</w>") else x[: -len("</w>")] for x in targets]
```

Comment:
First, note that in this case, `xxx</w>` (`xxx` is just a symbol here) is actually a single token in the vocabulary, but decoding it (done in the test) gives `xxx`


> * Option 1: The `targets` are wrong. Since `decode(encode(target)) != target` (which is OK). Maybe we should use `targets = [decode(encode(t)) for t in targets]` within the test.

I guess you mean setting `targets` to this new value, then pass it to the pipeline. If so, it's the right approach.
The current input value for `targets` to the pipeline is fine. This issue is coming from the fact we decode the output token id.

> Option 2: token_str is incorrect and ...
I am not really sure it is really worth the effort considering we clearly know (and even accept) the fact `decode(encode(target)) != target`. And how doable this approach.

> * Option 3: We need to send the current `targets` because they are "correct" for the pipeline, however, we need to filter them with `decode` before testing to align with the pipeline's work

I think what I am doing so far in this PR is already this Option 3, if I understand your word correctly. Although my change is a bit too specific (usage of `</w>` etc.)

I will try to make it more general.